### ISSUE TITLE
Typescript issues when no 'dom' in tsconfig

### ISSUE THREAD

**Issue by DanMossa** [dmester/jdenticon#38]
Hello all!
I'm using jdenticon on a backend node server and therefore have no need for the "dom" inside my "lib" in the `tsconfig.json`

The error I'm getting is

```
node_modules/jdenticon/types/module.d.ts:116:43 - error TS2304: Cannot find name 'Element'.

116 export function update(elementOrSelector: Element | string, hashOrValue?: any, config?: JdenticonConfig | number): void;
                                              ~~~~~~~

node_modules/jdenticon/types/module.d.ts:132:49 - error TS2304: Cannot find name 'Element'.

132 export function updateCanvas(elementOrSelector: Element | string, hashOrValue?: any, config?: JdenticonConfig | number): void;
                                                    ~~~~~~~

node_modules/jdenticon/types/module.d.ts:148:46 - error TS2304: Cannot find name 'Element'.

148 export function updateSvg(elementOrSelector: Element | string, hashOrValue?: any, config?: JdenticonConfig | number): void;
```

**Comment by DanMossa** [comment]
Looking at the `module.d.ts` I see the three functions, `update`, `updateCanvas` and `updateSvg` and they all have the comments 

> * This method is only available in the browser. Calling this method on Node.js will throw an error.

Is it possible to make these optional or something?

**Comment by dmester** [comment]
Thanks for reporting this issue! The module typings are shared between the browser and Node implementation of Jdenticon so unfortunately those functions will need to remain in the typings to support people importing the module into their web projects.

I researched how people have solved similar issues on DefinitelyTyped, and I found mainly two approaches:
* Add a dummy interface for `Element`. If the consumer of Jdenticon has `dom` added to `lib` in their `tsconfig.json`, the interface will be merged with the real one.
* Add `/// <reference lib="dom"/>` into the typings, but it has the side-effect of exposing the entire `dom` lib to the consumer of Jdenticon.

I just released a patch to NPM containing the first approach. Just update to version 3.0.1 and the compilation errors should hopefully go away.
